jdi-light/src/main/java/com/epam/jdi/light/actions/ActionObject.java

```diff
@@ -11,36 +12,56 @@
 import java.util.List;
 
 import static com.epam.jdi.light.actions.ActionHelper.*;
-import static com.epam.jdi.light.actions.ActionOverride.*;
-import static com.epam.jdi.light.settings.JDISettings.*;
-import static com.epam.jdi.tools.ReflectionUtils.*;
+import static com.epam.jdi.light.actions.ActionOverride.getOverrideAction;
+import static com.epam.jdi.light.settings.JDISettings.getJDISettings;
+import static com.epam.jdi.tools.ReflectionUtils.isInterface;
 
 public class ActionObject {
+    private final JDISettings jdiSettings = getJDISettings();
+    private ProceedingJoinPoint jp;
+    private final CacheValue<Object> obj = new CacheValue<>(
+            () -> jp.getThis() != null ? jp.getThis() : new Object());
+    private final String className;
+    public String stepUId = "";
+    private int elementTimeout;
+    private final CacheValue<Integer> timeout = new CacheValue<>(this::getTimeout);
+    private final CacheValue<JFunc1<Object, Object>> overrideAction = new CacheValue<>(this::getOverride);
+    private static final Safe<List<String>> isOverride = new Safe<>(ArrayList::new);
+
     public ActionObject(ProceedingJoinPoint joinPoint, String className) {
         this.jp = joinPoint;
         this.className = className;
         try {
             this.elementTimeout = element() != null
-                ? element().base().getTimeout()
-                : TIMEOUTS.element.get();
+                    ? element().base().getTimeout()
+                    : jdiSettings.TIMEOUTS.element.get();
         } catch (Throwable ex) {
             this.elementTimeout = 10;
         }
     }
-    public ProceedingJoinPoint jp() { return jp; }
-    private ProceedingJoinPoint jp;
-    public String className() { return className; }
-    private String className;
-    public String stepUId = "";
+
+    public ProceedingJoinPoint jp() {
+        return jp;
+    }
+
+    public String className() {
+        return className;
+    }
+
     public boolean topLevel() {
         return aroundCount(className()) == 1;
     }
-    public Object object() { return obj.get(); }
-    private CacheValue<Object> obj = new CacheValue<>(
-        () -> jp.getThis() != null ? jp.getThis() : new Object());
 
-    public IBaseElement element() { return element.get(); }
-    private CacheValue<IBaseElement> element = new CacheValue<>(this::getElement);
+    public Object object() {
+        return obj.get();
+    }
+
+    public IBaseElement element() {
+        return element.get();
+    }
+
+    private final CacheValue<IBaseElement> element = new CacheValue<>(this::getElement);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5482899)